.github/workflows/continuous-integration.yaml

```diff
@@ -0,0 +1,146 @@
+name: Continuous Integration
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+  push:
+    branches:
+      - master
+
+jobs:
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Cache npm dependencies
+        uses: actions/cache@v1
+        with:
+          key: npm-${{ hashFiles('**/package-lock.json') }}
+          path: ~/.npm
+          restore-keys: |
+            npm-
+      - name: Install dependencies
+        run: npm ci --ignore-scripts --no-audit --prefer-offline --progress=false
+      - name: Format
+        run: npm run format
+  lint-javascript:
+    name: Lint JavaScript
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Cache npm dependencies
+        uses: actions/cache@v1
+        with:
+          key: npm-${{ hashFiles('**/package-lock.json') }}
+          path: ~/.npm
+          restore-keys: |
+            npm-
+      - name: Cache local cache
+        uses: actions/cache@v1
+        with:
+          key: eslint-${{ github.sha }}
+          path: .cache
+          restore-keys: |
+            eslint-
+      - name: Install dependencies
+        run: npm ci --ignore-scripts --no-audit --prefer-offline --progress=false
+      - name: Lint
+        run: |
+          # This is required to workaround a limitation of ESLint which uses
+          # file modification time in its cache, instead of file checksum.
```

Comment:
Could we disable the ESLint caching in CI instead? This seems a bit complicated and it doesn’t seem like there is much of a win. It seems easy enough to check lint every time.

Reply:
I've seen projects where ESLint takes 10+ minutes, so I tend to add it. But here it seems quite quick, so I removed the cache. We can always get back to it in the future if we need to :) eeba69e047ece013db42cdba827cdca003ffa772

Reply:
Wow, 10 minutes!